### PR TITLE
Use php included JSON pretty print

### PR DESCRIPTION
**Pull request by Hipska**
Note that I added deprecation warning to next version release as this will bump minimal required php version to 5.4.0

### CODE REVIEW

**Comment by piRGoif** [comment]
Seems like a good idea, thanks ! Seems like the \Utils::JSONPrettyPrint method is only used once in Combodo's collectors, but indeed adding a `@deprecated` is a really good idea as this will allow other consumers to migrate !

File: core/utils.class.inc.php [hunk 1]
```diff
@@ -439,6 +439,7 @@ static public function DoPostRequest($sUrl, $aData, $sOptionnalHeaders = null, &
 
 	/**
 	 * Pretty print a JSON formatted string. Copied/pasted from http://stackoverflow.com/questions/6054033/pretty-printing-json-with-php
+	 * @deprecated 1.3.0
```
Comment (piRGoif) [inline core/utils.class.inc.php:442]: When possible it's nice to explain what to use instead O:)

```suggestion
	 * @deprecated 1.3.0 use `json_encode($json, JSON_PRETTY_PRINT);` instead
```

**Review by piRGoif — CHANGES_REQUESTED** [review]

**Comment by Hipska** [comment]
@piRGoif What about replacing the content of the deprecated method with a php native code?

**Review by piRGoif — APPROVED** [review]

**Comment by Hipska** [comment]
And maybe also a log warning message?

**Comment by piRGoif** [comment]
Thanks for the deprecated message change !

About changing \Utils::JSONPrettyPrint, I would advise against it as this may introduce behavior change : both codes (Combodo's and PHP) might not produce the same result...
Having a transition period will allow to fallback to the old method if needed.

I'm setting this PR to be reviewed by the dev team !

**Comment by Hipska** [comment]
Good point, but I would still add a log message if this method is still used. Agree?

**Comment by piRGoif** [comment]
> And maybe also a log warning message?

Yes seems like a good idea.
I have an ungoing proposition about logging deprecated calls (Combodo/iTop#193) but it won't apply here as this would introduce dependency to the iTop version where this will be introduced (probably 3.0.0).
So a trigger_error maybe ? We'll have to talk about this with the dev team !

**Comment by Hipska** [comment]
Oh yeah, I was just thinking about a simple `Utils::Log` message.

**Comment by piRGoif** [comment]
>> a trigger_error maybe ?
> Oh yeah, I was just thinking about a simple `Utils::Log` message.

That is an option too indeed, and less "violent". Well noted !

**Review by piRGoif — APPROVED** [review]
It was a good idea to add a notice example. Again, the final solution will be chosed by the dev team
I'm approving the code as is for now, and we'll see what the dev team discussion brings to the table :)

**Comment by piRGoif** [comment]
Ref for `JSON_PRETTY_PRINT` flag availability : https://www.php.net/manual/fr/json.constants.php#constant.json-pretty-print

File: core/utils.class.inc.php [hunk 2]
```diff
@@ -439,11 +439,14 @@ static public function DoPostRequest($sUrl, $aData, $sOptionnalHeaders = null, &
 
 	/**
 	 * Pretty print a JSON formatted string. Copied/pasted from http://stackoverflow.com/questions/6054033/pretty-printing-json-with-php
+	 * @deprecated 1.3.0 use `json_encode($value, JSON_PRETTY_PRINT);` instead
```
Comment (piRGoif) [inline core/utils.class.inc.php:442]:
```suggestion
	 * @deprecated 1.3.0 use `json_encode($value, JSON_PRETTY_PRINT);` instead (PHP 5.4.0 required)
```

**Review by piRGoif — CHANGES_REQUESTED** [review]
Just reviewing this PR before showing it to the dev team, I think adding the PHP requirement in the phpdoc would be a good idea

**Review by piRGoif — APPROVED** [review]
Thanks !

File: toolkit/dump_tasks.php [hunk 3]
```diff
@@ -113,8 +113,7 @@
 					$aCurrentTaskDefinition['database_table_name'] = "";
 				}
 
-				$sDefinition = json_encode($aCurrentTaskDefinition);
-				echo Utils::JSONPrettyPrint($sDefinition)."\n";
+				echo json_encode($aCurrentTaskDefinition, JSON_PRETTY_PRINT);
```
Comment (piRGoif) [inline toolkit/dump_tasks.php:116]:
```suggestion
				if (defined('JSON_PRETTY_PRINT') {
					echo json_encode($aCurrentTaskDefinition, JSON_PRETTY_PRINT);
				} else {
					echo Utils::JSONPrettyPrint($sDefinition)."\n";
				}
```

**Comment by piRGoif** [comment]
Thanks, merging !

**Comment by Hipska** [comment]
FYI, i don't think there is still the need to check if the pretty print exists as the collector requires 5.6 minimal by default..

**Comment by piRGoif** [comment]
Thanks for this remark
I'm not sure the collector does a good job checking for PHP Version ? 
This would take some time to check, I'm not sure this is worse the effort ?

**Comment by Hipska** [comment]
The problem is more that the json/csv/sql collectors are loaded/included always. And they set their own php version requirement, even when they are not used. The command used is just `phpversion()`, so I don't think that's giving a impact.

The `Orchestrator::AddCollector` could call a method on the class like `Collector::GetRequirementList` maybe?

**Pull request merged**